### PR TITLE
Add server.jmx.port option to start JMX agent

### PR DESCRIPTION
**Pull request by frsyuki**
Main goal is to get GC status using JMX. It's also possible to export
custom metrics by adding `@Managed` to an injected class and
exposing the class using `ExportBinder.newExporter` method.

### CODE REVIEW

**Comment by frsyuki** [comment]
@komamitsu PTAL

File: digdag-server/src/main/java/io/digdag/server/ServerConfig.java [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
     public Optional<String> getAccessLogPath();
 
+    public Optional<Integer> getJmxPort();
```
Comment (komamitsu) [inline digdag-server/src/main/java/io/digdag/server/ServerConfig.java]: [minor] `public` isn't needed here.

File: digdag-tests/src/test/java/acceptance/ServerJmxIT.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import javax.management.MBeanServerConnection;
+import javax.management.ObjectName;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.Assert.assertThat;
+
+public class ServerJmxIT
+{
+    public static final Pattern JMX_PORT_PATTERN = Pattern.compile("\\s*JMX agent started on port (\\d+)\\s*");
```
Comment (komamitsu) [inline digdag-tests/src/test/java/acceptance/ServerJmxIT.java]: [trivial] We can set it to`private`

**Comment by komamitsu** [comment]
👍

**Pull request merged**